mubench.reviewsite/src/DBConnection.php

```diff
@@ -210,6 +213,22 @@ public function getPotentialHits($experiment, Detector $detector, $project_id, $
             ->orderBy($this->query_builder->raw("`rank` * 1"))->get();
     }
 
+    private function getViolationTypeNamesForMisuse($project, $version, $misuse)
+    {
+        $type_ids = $this->table('misuse_types')->select('type')->where('project', $project)->where('version', $version)->where('misuse', $misuse)->get();
```

Comment:
Doesn't the DB library support joins? This could be one query "SELECT name FROM misuse_types, types WHERE project = ... AND version = ... AND misuse = ... AND type = id".